source/views/student-orgs/detail.ios.js

```diff
@@ -16,128 +18,143 @@ const styles = StyleSheet.create({
 
   render() {
-    let knownData = this.props.base
-    let orgName = knownData.name.trim()
-    let orgCategory = knownData.categories.join(', ')
+    const data = this.props.org
+    const name = data.name.trim()
+    let {
+      category,
+      meetings,
+      contacts,
+      description,
+      advisors,
+      lastUpdated,
+      website,
+    } = data
 
-    let contents
-    if (!this.props.loaded) {
-      contents = (
-        <Section header="ORGANIZATION">
-          <Cell cellStyle="Basic" title="Loading…" />
-        </Section>
-      )
-    } else if (!this.props.full) {
-      contents = (
-        <Section header="ORGANIZATION">
-          <Cell cellStyle="Basic" title="No information found." />
-        </Section>
-      )
-    } else {
-      contents = this.renderBody(this.props.full)
-    }
+    advisors = advisors.filter(c => c.name.trim().length)
 
     return (
       <ScrollView>
         <TableView>
-          <Text selectable={true} style={styles.name}>{orgName}</Text>
-
-          <Section header="CATEGORY">
-            <Cell cellStyle="Basic" title={orgCategory} />
+          <Text selectable={true} style={styles.name}>{name}</Text>
+
+          {category.trim()
+            ? <Section header="CATEGORY">
+                <Cell cellStyle="Basic" title={category} />
+              </Section>
+            : null}
+
+          {meetings.trim()
+            ? <Section header="MEETINGS">
+                <Cell
+                  cellContentView={
+                    <Text style={styles.meetings}>{meetings}</Text>
+                  }
+                  cellStyle="Basic"
+                />
+              </Section>
+            : null}
+
+          {website.trim()
+            ? <Section header="WEBSITE">
+                <Cell
+                  cellStyle="Basic"
+                  accessory="DisclosureIndicator"
+                  title={website}
+                  onPress={() => openUrl(`http://${website}`)}
+                />
+              </Section>
+            : null}
+
+          <Section header="CONTACT">
+            {contacts.map((c, i) => (
+              <Cell
+                key={i}
+                cellStyle={c.title ? 'Subtitle' : 'Basic'}
+                accessory="DisclosureIndicator"
+                title={`${c.firstName.trim()} ${c.lastName.trim()}`}
+                detail={c.title.trim()}
+                onPress={() => Linking.openURL(`mailto:${c.email}@stolaf.edu`)}
+              />
+            ))}
           </Section>
 
-          {contents}
-          {this.displayFooter()}
+          {advisors.length
+            ? <Section header={advisors.length === 1 ? 'ADVISOR' : 'ADVISORS'}>
+                {advisors.map((c, i) => (
+                  <Cell
+                    key={i}
+                    cellStyle="Basic"
+                    accessory="DisclosureIndicator"
+                    title={c.name.trim()}
+                    onPress={() => Linking.openURL(`mailto:${c.email}`)}
```

Comment:
Student's emails are returned as just the username, while advisor's are the full name+domain.